### PR TITLE
proxy server initial set up

### PR DESCRIPTION
**Pull request by LittleJennie**

### CODE REVIEW

File: dist/index.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    <div id="menu"></div>
+    <div id="reviews"></div>
+    <footer>Contributors: Minho Choo, Jennie Zeng, Priyanka Pramanik </footer>
+
+    <script src="http://localhost:3002/bundle.js"></script>
+    <script src="http://localhost:3003/main.js"></script>
+    <script src="http://localhost:3004/bundle.js"></script>
+
+  </body>
+</html>
```
Comment (jonathantch) [inline dist/index.html]: Need additional line at the end of the file. Apply here and elsewhere.

File: server/app.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  console.log('redirecting to Reservations');
+  apiProxy.web(req, res, {target: reservations});
+});
+
+app.all("/restaurants/:restaurant_id/reviews", function(req, res) {
+  console.log('redirecting to Reviews');
+  apiProxy.web(req, res, {target: reviews});
+});
+
+app.listen(3000);
```
Comment (jonathantch) [inline server/app.js]: Make line 49 a callback of app.listen. Apply here and elsewhere.

**Review by jonathantch — COMMENTED** [review]
Suggest to put package-lock.json in .gitignore. Maybe make a todo to update ES5 function syntax to arrow function.

File: dist/index.html (hunk 1 above)
Comment (LittleJennie) [reply dist/index.html]: extra line added in new commit

File: server/app.js (hunk 2 above)
Comment (LittleJennie) [reply server/app.js]: extra line added in new commit

File: server/app.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+const menus = 'http://localhost:3002';
+const reservations = 'http://localhost:3003';
+const reviews = 'http://localhost:3004';
+
+app.set('port', 3000);
+
+app.use(morgan('dev'));
+app.use('/:restaurantId', express.static(path.resolve('dist')));
+
+app.all('/menuData/:restaurant_id', (req, res) => {
```
Comment (primanbo) [inline server/app.js:22]:
```suggestion
app.all('/menuData/:restaurantId', (req, res) => {
```

File: server/app.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+app.all('/targettimeslots/:restaurantId', (req, res) => {
+  apiProxy.web(req, res, {target: reservations});
+});
+
+app.all('/reservations/:restaurantId', (req, res) => {
+  apiProxy.web(req, res, {target: reservations});
+});
+
+app.all('/restaurants/:restaurant_id/reviews', (req, res) => {
```
Comment (primanbo) [inline server/app.js:46]:
```suggestion
app.all('/restaurants/:restaurantId/reviews', (req, res) => {
```

File: .gitignore [hunk 5]
```diff
@@ -0,0 +1,4 @@
+node_modules
+menu
+reservations
+reviews
```
Comment (primanbo) [inline .gitignore:4]: Add new line at end of file
```suggestion
reviews

```

**Review by primanbo — COMMENTED** [review]
Looks good, @LittleJennie!

Just check that the restaurantId case is consistent in your API server routes before merging.

File: server/app.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+const app = express();
+const apiProxy = httpProxy.createProxyServer();
+
+const menus = 'http://localhost:3002';
+const reservations = 'http://localhost:3003';
+const reviews = 'http://localhost:3004';
+
+app.set('port', 3000);
+
+app.use(morgan('dev'));
```
Comment (primanbo) [inline server/app.js:19]: Such a good idea to use morgan!

**Pull request merged**